src/segment_wifi.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"regexp"
+	"strings"
+)
+
+type wifi struct {
+	props              *properties
+	env                environmentInfo
+	State              string
+	SSID               string
+	RadioType          string
+	AuthenticationType string
+	Channel            string
+	ReceiveRate        string
+	TransmitRate       string
+	SignalStrength     string
+}
+
+const (
+	State              = "State"
+	SSID               = "SSID"
+	RadioType          = "Radio type"
+	AuthenticationType = "Authentication"
+	Channel            = "Channel"
+	ReceiveRate        = "Receive rate (Mbps)"
+	TransmitRate       = "Transmit rate (Mbps)"
+	SignalStrength     = "Signal"
+)
+
+func (w *wifi) enabled() bool {
+	// If in Linux, who wnows. Gonna need a physical linux machine to develop this since the wlan interface isn't available in wsl
+	// But possible solution is to cat /proc/net/wireless, and if that file doesn't exist then don't show the segment
+	// In Windows and in wsl you can use the command `netsh.exe wlan show interfaces`. Note the .exe is important for this to worw in wsl.
+	if w.env.getPlatform() == windowsPlatform || w.env.isWsl() {
+		cmd := "netsh"
+		if !w.env.hasCommand(cmd) {
+			return false
+		}
+		cmdResult, err := w.env.runCommand(cmd, "wlan", "show", "interfaces")
+		displayError := w.props.getBool(DisplayError, false)
+		if err != nil && displayError {
+			w.State = "WIFI ERR"
+			// w.Namespace = w.Context
+			return true
+		}
+		if err != nil {
+			return false
+		}
+
+		regex := regexp.MustCompile(`(.+) : (.+)`)
```

Comment:
we also have regex helpers for this (we cache the compilations), see `regex.go` for more information.

Reply:
Oh nice. I will look into that and update my other PR.